docs/configuration-management.md

```diff
@@ -0,0 +1,110 @@
+* **Caching**
+  * A cache entry is the merge result of multiple buckets and specific to one particular cluster.
+  * It has to be possible to identify in which cache entry a specific configuration value was used.
+  * If a configuration value has changed (other attributes coupled to a configuration entry don't matter), any cache entries which were using this configuration value have to be invalidated.
+  * The invalidation of a cache entry leads to a reconciliation for the related clusters (scheduling of a reconciliation run).
+* **CLI**
+  * A CLI is used to access the configuration management
+  * The CLI commands have following syntax: `cmd <verb> <nome> [-flag1] [-flag2=valueOfFlag] ...`.
+  
+    Example: `kyco list buckets`,  `kyco add entry -b bucket -k key -v value -t integer` etc.
+
+## Solution Strategy
+
+### Data model
+
+The data of the configuration management can be distinguished in two domains:
+
+* Configuration data
+
+  Any information related to the configuration entires (metadata per configuration key, different versions of configuration entries etc.).
+
+* Caching data
+
+  A cache entity contains the merged configuration for one cluster. It also trackes which configuration entry was used in which cache entity to identify outdated cluster configurations.
+
+#### Configuration Data
+
+Requirements for the data structure layout:
+
+* Store key and values entities in different versions.
+* Map a value entry to a key entry.
+
+**Configuration key table:**
+
+|Column|Description|Data Type|Primary Key|Example|
+|--|--|--|--|--|
+|key|The configuration key|string|Yes|`my.config.key`|
+|version|Incremental version number|Integer|Yes|`1`|
+|datatype|Data type of the stored information|String|No|`integer`|
+|encrypted|Flag whether the value is plain text or encrypted|Boolean|No|`false`|
+|created|Timestamp when the entry was created|Integer|No|`123456789`|
+|user|User who created the entry|String|No|`i98765`|
+|validator|Optional logic which is executed to validated the value. Return value has to be a boolean: `true`=valid / `false`=invalid|String|No|`it >= 1 && it < 10`
+|trigger|Optional trigger functions which are executed by the reconciler|String|No|`restartPod("app=appLabel,component=abc")`
+
+**Configuration value table:**
+
+|Column|Description|Data Type|Primary Key|Example|
+|--|--|--|--|--|
+|key|Foreign key to configuration key|String|Yes|`my.config.key`|
+|key_version|Foreign key to configuration key version|Integer|No|`1`|
+|version|Incremental version number|Integer|No|`1`|
+|bucket|The bucket the configuration entry belongs to|String|Yes|`cust1`|
+|value|The value of the configuration entry|String|No|`123`|
+|created|Timestamp when the entry was created|Integer|No|`123456789`|
+|user|User who created the entry|String|No|`i98765`|
+
+#### Cache Table
+
+Requirements for the data structure layout:
+
+* Cache bucket merge result per cluster.
```

Comment:
Why is this one not a sentence! All the others are full sentences! :O

Reply:
Hm, good point. Will fix that when I'm updating the tech-docs in general. This file contains currently only some basic information.